### PR TITLE
Rdp server

### PR DESCRIPTION
**Pull request by arrooney**
## Description of changes
These changes split up the central server and has RDP sockets for each service seperately. The advantages are:
* Don't need queues to manage packets, since this is implemented in CSP
* Don't need a 'response server', since the responses will be sent on the same connection they came in on which simplifies data flow
* Acknowledgements, timeouts and handshakes are taken care of with RDP: these sockets will only connect VIA RDP to ensure ordering and integrity
* Because of these things, there is less CPU overhead per transaction

## Testing done

Sent and received packets on all services with the redesigned ground station, all tests passed.

TODO: send failure packets in all failure cases of the services.

## Merge checklist
- [x] Docker image or CCS Project builds with the changes
- [x] Docker image or CCS Project runs with the changes
- [x] Can send commands from a mocked ground station/relevant changes tested
- [ ] Changes have been reviewed

### CODE REVIEW

File: Services/service_response.c
Comment (arrooney) [inline Services/service_response.c:23]: Left this file in just because it might be useful in the future.

**Comment by ArashYzd** [comment]
Looks good to me. Do you want my opinion on a specific part or do you want me to do some kind of testing?

**Comment by arrooney** [comment]
> Looks good to me. Do you want my opinion on a specific part or do you want me to do some kind of testing?

Just wanted a once-over to have another pair of eyes on the changes.

We won't merge until the ground station changes are ready

**Comment by arrooney** [comment]
@TheCandianVendingMachine I havent tested this with the demo server. I think it should work, but there may be some stray function calls to non-existent functions in main.

**Comment by TheCandianVendingMachine** [comment]
> @TheCandianVendingMachine I havent tested this with the demo server. I think it should work, but there may be some stray function calls to non-existent functions in main. Also just FYI since these changes might effect your work.

If it doesn't affect system functionality I should be fine, I am not merging master changes into my branch. Future work maybe though, that's a future me problem though

**Review by ArashYzd — APPROVED** [review]

**Pull request merged**
